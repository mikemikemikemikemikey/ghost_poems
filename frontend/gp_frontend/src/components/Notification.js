import React from 'react'

const Notification = ({ message }) => {
  const errorStyle = {
    color: 'red',
    background: 'lightgrey',
    fontSize: 20,
    borderStyle: 'solid',
    borderRadius: 5,
    padding: 10,
    paddingBottom: 14,
    marginTop: 8,
    marginBottom: 0,
  }

  if (!message.error) errorStyle.color = 'green'
  if (message.message === null) {
    return null
  }

  return (
    <div style={errorStyle} className="error" data-cy='message'>
      {message.message}
    </div>
  )
}

export default Notification
