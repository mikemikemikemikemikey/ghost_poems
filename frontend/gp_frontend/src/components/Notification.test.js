import React from 'react'
import Notification from './Notification'
import  renderer from 'react-test-renderer'

it('renders correctly when error', () => {
    const message = {
        message: 'test message error',
        error: true,
    }
    const tree = renderer
        .create(<Notification message={message} />)
        .toJSON()

    expect(tree).toMatchSnapshot()
})

it('renders correctly when success', () => {
    const message = {
        message: 'test message success',
        error: false,
    }
    const tree = renderer
        .create(<Notification message={message} />)
        .toJSON()
        
    expect(tree).toMatchSnapshot()
})