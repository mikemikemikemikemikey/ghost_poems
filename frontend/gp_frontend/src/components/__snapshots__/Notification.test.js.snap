// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when error 1`] = `
<div
  className="error"
  data-cy="message"
  style={
    Object {
      "background": "lightgrey",
      "borderRadius": 5,
      "borderStyle": "solid",
      "color": "red",
      "fontSize": 20,
      "marginBottom": 0,
      "marginTop": 8,
      "maxHeight": 20,
      "padding": 10,
      "paddingBottom": 14,
    }
  }
>
  test message error
</div>
`;

exports[`renders correctly when success 1`] = `
<div
  className="error"
  data-cy="message"
  style={
    Object {
      "background": "lightgrey",
      "borderRadius": 5,
      "borderStyle": "solid",
      "color": "green",
      "fontSize": 20,
      "marginBottom": 0,
      "marginTop": 8,
      "maxHeight": 20,
      "padding": 10,
      "paddingBottom": 14,
    }
  }
>
  test message success
</div>
`;
